extensions/sql-migration/src/api/azure.ts

```diff
@@ -510,16 +566,140 @@ export async function getLocationDisplayName(location: string): Promise<string>
 	return api.getRegionDisplayName(location);
 }
 
+export async function validateIrSqlDatabaseMigrationSettings(
+	migration: MigrationStateModel,
+	sourceServerName: string,
+	trustServerCertificate: boolean,
+	sourceDatabaseName: string,
+	targetDatabaseName: string,
+	testIrOnline: boolean = true,
+	testSourceConnectivity: boolean = true,
+	testTargetConnectivity: boolean = true): Promise<ValdiateIrDatabaseMigrationResponse> {
+
+	const api = await getAzureCoreAPI();
+	const account = migration._azureAccount;
+	const subscription = migration._targetSubscription;
+	const serviceId = migration._sqlMigrationService?.id;
+	const host = api.getProviderMetadataForAccount(account).settings.armResource?.endpoint;
+	const path = encodeURI(`${serviceId}/validateIr?api-version=${DMSV2_API_VERSION}`);
+	const targetDatabaseServer = migration._targetServerInstance as AzureSqlDatabaseServer;
+
+	const requestBody: ValidateIrSqlDatabaseMigrationRequest = {
+		sourceDatabaseName: sourceDatabaseName,
+		targetDatabaseName: targetDatabaseName,
+		kind: AzureResourceKind.SQLDB,
+		validateIntegrationRuntimeOnline: testIrOnline,
+		sourceSqlConnection: {
+			testConnectivity: testSourceConnectivity,
+			dataSource: sourceServerName,
+			userName: migration._sqlServerUsername,
```

Comment:
I'm not sure if this is the right request body to be leaving this comment on (maybe it's the one after this), but I filed a bug regarding a 400 error in the validation request due to a problematic `username` argument: https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/2126193